nservicebus/messaging/routing.md

```diff
@@ -0,0 +1,165 @@
+---
+title: Message routing
+summary: How NServiceBus routes messages between the endpoints
+tags:
+- routing
+- message
+- route
+- send
+- publish
+- reply
+---
+
+One of the things NServiceBus takes care of is the routing of messages. Usually the only thing develop has to do is call `Send`, `Publish` or `Reply` and the actual message destination is calculated by the framework.
+
+## Endpoint and endpoint instances
+
+Endpoint is a logical concept that relates to a program that uses NServiceBus to communicate with other similar programs. Each endpoint has a name e.g. `Sales` or `OrderProcessing`.
+
+During deployment each endpoint might be materialized in form of one or many instances. Each instance is an identical copy of binaries resulting from building the endpoint program code base. Usually each endpoint instance is placed in separate directory or on separate machine. Each instance has its unique name that consists of the name of the endpoint (e.g. `Sales`) and up to two *discriminator* values:
```

Comment:
Does this mean both user-provided and transport provided discriminators can be combined together?